### PR TITLE
#2824 Fix stale images script

### PR DESCRIPTION
**Pull request by christian-kreuzberger-dtx**
Signed-off-by: Christian Kreuzberger <christian.kreuzberger@dynatrace.com>

* This PR fixes the failed workflows for stale and outdated images
* Set OS to ubuntu-20.04 (instead of `ubuntu-latest`)
* added c-py/action-dotenv-to-setenv@v2 for loading .ci_env
* added more (older) services for outdated tag check
* added a check for images/tags based on commit hashes that are outdated
* added some more debug messages
* added sanity checks

In addition, I've ran the script locally and cleaned up old/outdated-images on `keptn` organization on hub.docker.com.

### CODE REVIEW

File: .github/workflows/cleanup_images.yml [hunk 1]
```diff
@@ -6,13 +6,20 @@ on:
...
+        with:
+          env-file: .ci_env
       - name: Clean up outdated images
         env:
           REGISTRY_USER: ${{ secrets.REGISTRY_USER }}
           REGISTRY_PASSWORD: ${{ secrets.REGISTRY_PASSWORD }}
+          DOCKER_ORG: ${{ env.DOCKER_ORGANIZATION }}
         run: |
-          apt-get update && apt-get install jq -y && ./gh-actions-scripts/cleanup_docker_images.sh
+          # sudo apt-get update && sudo apt-get install jq -y
```
Comment (bacherfl) [inline .github/workflows/cleanup_images.yml:24]: Is this commented out intentionally?

File: .github/workflows/detect_stale_images.yml [hunk 2]
```diff
@@ -10,9 +10,16 @@ jobs:
...
+        with:
+          env-file: .ci_env
       - name: Detect stale images on DockerHub
         env:
           REGISTRY_USER: ${{ secrets.REGISTRY_USER }}
           REGISTRY_PASSWORD: ${{ secrets.REGISTRY_PASSWORD }}
+          DOCKER_ORG: ${{ env.DOCKER_ORGANIZATION }}
         run: |
-          apt-get update && apt-get install jq -y && ./gh-actions-scripts/detect_stale_images.sh
+          # sudo apt-get update && sudo apt-get install jq -y
```
Comment (bacherfl) [inline .github/workflows/detect_stale_images.yml:24]: Is this commented out intentionally?

File: .github/workflows/cleanup_images.yml (hunk 1 above)
Comment (christian-kreuzberger-dtx) [reply .github/workflows/cleanup_images.yml:24]: Yes, it looks like those tools are installed automatically on ubuntu 20.04. I'll remove the line.

Comment (christian-kreuzberger-dtx) [reply .github/workflows/cleanup_images.yml]: Line removed.

File: .github/workflows/detect_stale_images.yml (hunk 2 above)
Comment (christian-kreuzberger-dtx) [reply .github/workflows/detect_stale_images.yml]: Line removed.

File: gh-actions-scripts/cleanup_docker_images.sh [hunk 3]
```diff
@@ -62,27 +93,23 @@ for s in ${IMAGES[@]}; do
   # get all outdated tag where tag contains "bug"
   outdated_bug_tags=$(get_outdated_images $s "bug" $MAX_AGE)
   # get all outdated tag where tag contains "patch"
-  outdated_patch_tags=$(get_outdated_images $s "patch" $MAX_AGE)
+  # outdated_patch_tags=$(get_outdated_images $s "patch" $MAX_AGE)
```
Comment (warber) [inline gh-actions-scripts/cleanup_docker_images.sh:96]: is this line disabled intentionally?

Comment (christian-kreuzberger-dtx) [reply gh-actions-scripts/cleanup_docker_images.sh:96]: good catch. Yes it's disabled intentionally, but I should also disable the for loop below.

File: gh-actions-scripts/cleanup_docker_images.sh [hunk 4]
```diff
@@ -30,59 +93,90 @@ function get_outdated_images() {
...
+  # get all outdated commit hash tags
+  outdated_commit_hash_tags=$(get_outdated_commit_hash_tags $s $MAX_AGE)
+
+  outdated_datetime_tags=$(get_outdated_datetime_tags $s "2020" $MAX_AGE)
+
+  # get all outdated tag where tag contains "dev-PR"
+  # outdated_dev_pr_tags=$(get_outdated_images $s "dev-PR" $MAX_AGE)
+
+  # ToDo: Also Check for "x.y.z-dev.20" tags (e.g., 0.8.0-dev.20210101)
+  # outdated_dev_tags=$(get_outdated_images $s "dev.20" $MAX_AGE)
```
Comment (christian-kreuzberger-dtx) [inline gh-actions-scripts/cleanup_docker_images.sh:139]: commented out on purpose for now

File: gh-actions-scripts/cleanup_docker_images.sh [hunk 5]
```diff
@@ -30,59 +93,90 @@ function get_outdated_images() {
...
+    delete_tag $s $tag
+  done
+
+  for tag in ${outdated_dev_pr_tags}; do
+    echo "dummy" $s $tag
+  done
+
+  for tag in ${outdated_dev_tags}; do
+    echo "dummy" $s $tag
+  done
```
Comment (christian-kreuzberger-dtx) [inline gh-actions-scripts/cleanup_docker_images.sh:165]: no real delete here for now, just a dummy output

**Review by bacherfl — APPROVED** [review]
lgtm

**Pull request merged**